docs/contributing.md

```diff
@@ -0,0 +1,115 @@
+
+This repository relies on Yarn's [workspaces feature](https://yarnpkg.com/features/workspaces) to provide a way to work with packages individually and collectively. Refer to the documentation for the following Yarn commands for more information:
+
+- [`yarn workspace`](https://yarnpkg.com/cli/workspace)
+- [`yarn workspaces foreach`](https://yarnpkg.com/cli/workspaces/foreach)
+
+> **Note**  
+> `workspaceName` in the Yarn documentation is the `name` field within a package's `package.json`, e.g., `@metamask/announcement-controller`, not the directory where it is located, e.g., `packages/announcement-controller`.
+
+## Releasing
+
+The [`create-release-branch`](https://github.com/MetaMask/create-release-branch) tool and [`action-publish-release`](https://github.com/MetaMask/action-publish-release) GitHub action are used to automate the release process.
+
+1. **Create a release branch.**
+
+   Run `yarn create-release-branch`. This tool generates a file and opens it in your editor, where you can specify which packages you want to include in the next release and which versions they should receive. Instructions are provided for you at the top; read them and update the file accordingly.
+
+   When you're ready to continue, save and close the file.
+
+2. **Update changelogs for relevant packages.**
+
+   At this point you will be on a new release branch, and a new section will have been added to the changelog of each package you specified in the previous step.
+
+   For each changelog, review the new section and make the appropriate changes:
+
+   - Move each entry into the appropriate category (review the ["Keep a Changelog"](https://keepachangelog.com/en/1.0.0/#types) spec for the full list of categories and the correct ordering of all categories).
+   - Remove any changelog entries that don't affect consumers of the package (e.g. lockfile changes or development environment changes). Exceptions may be made for changes that might be of interest despite not having an effect upon the published package (e.g. major test improvements, security improvements, improved documentation, etc.).
+   - Reword changelog entries to explain changes in terms that users of the package will understand (e.g., avoid referencing internal variables/concepts).
+   - Consolidate related changes into one change entry if it makes it easier to comprehend.
+
+   Run `yarn changelog:validate` to check that all changelogs are correctly formatted.
+
+   Commit and push the branch.
+
+3. **Submit a pull request for the release branch so that it can be reviewed and tested.**
+
+   Make sure the title of the pull request follows the pattern "Release \<new version\>".
+
+   If changes are made to the base branch, the release branch will need to be updated with these changes and review/QA will need to restart again. As such, it's probably best to avoid merging other PRs into the base branch while review is underway.
+
+4. **"Squash & Merge" the release.**
```

Comment:
Nit: Maybe worth advising that we double-check that the commit summary is correct here as well?

Reply:
Sure! Added in [bae8925](https://github.com/MetaMask/controllers/pull/993/commits/bae89258a3e9936fa0f6b3c4214a4f011cadd749).